pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -544,12 +546,23 @@ type testContext struct {
 
 func (tc *testContext) verify(t *testing.T, expected result, initialObjects []metav1.Object, result interface{}, status *framework.Status) {
 	t.Helper()
-	assert.Equal(t, expected.status, status)
+	if diff := cmp.Diff(expected.status, status, cmp.Comparer(func(s1, s2 *framework.Status) bool {
+		if s1 == nil && s2 == nil {
+			return true
+		}
+		if s1 == nil || s2 == nil {
+			return false
+		}
+		if s1.Code() != s2.Code() || s1.Message() != s2.Message() || s1.FailedPlugin() != s2.FailedPlugin() || s1.AsError().Error() != s2.AsError().Error() {
```

Comment:
I don't want to add a "one-off" solution.

We already have a definition of Equal for statuses in `pkg/scheduler/framework/interface.go`, which does error comparison based on `errors.Is`.

We either need to:
1. change the tests in a way that an error is another error (preferred, but probably requires more changes). One way to do this is to make sure that each error used implements `Is`. Another is to have the error as a variable that can be used when constructing the test cases.
2. Add `s1.AsError().Error() != s2.AsError().Error()` to the existing function. Although tests might start relying on it and it would be hard to remove it later.

Reply:
Thanks for the comment. I add a wrapper StatusError which will be the error type for all the statuses.

Reply:
Got the idea why we want to add the StatusError, but we call the `Equal` only in tests, right? I think we can handle this simply by only comparing the error messages, considering we only have two kind of errors now, one plain error and another fitError.